shaman/util.py

```diff
@@ -155,3 +156,25 @@ def parse_distro_query(query):
             dict(distro=distro, distro_codename=codename, distro_version=version, arch=arch)
         )
     return result
+
+
+def get_repo_url(query, project_name, ref, arch, directory=None, repo_file=True):
+    # requires the repository to be fully available on a remote chacra
+    # instance for a proper redirect. Otherwise it will fail explicitly
+    if arch == 'aarch64':
+        arch = 'arm64'
+    if arch == 'amd64':
+        arch = 'x86_64'
+    repo = query.filter_by(status='ready').first()
+    if arch:
+        repo = query.filter_by(status='ready').join(models.Arch).filter(models.Arch.name == arch).first()
+    if not repo:
+        abort(504, "no repository is ready for: %s/%s" % (project_name, ref))
+    repo_url = repo.url
+    if directory:
+        repo_url = os.path.join(repo.url, directory)
+    if repo_file:
+        # redirect to the repo endpoint in chacra that prints a plain text
```

Comment:
some other method/function redirects, not this one. This function is just providing a url. I'm being a stickler here, this isn't a hard request on my end, it just took me a bit to make sure there wasn't a redirect going on in that function